internal/admission/validator.go

```diff
@@ -157,10 +163,60 @@ func (validator KongHTTPValidator) ValidateConsumer(
 	return true, "", nil
 }
 
+func (validator KongHTTPValidator) ValidateConsumerGroup(
+	ctx context.Context,
+	consumerGroup kongv1beta1.KongConsumerGroup,
+) (bool, string, error) {
+	// Ignore ConsumerGroups that are being managed by another controller.
+	if !validator.ingressClassMatcher(&consumerGroup.ObjectMeta, annotations.IngressClassKey, annotations.ExactClassMatch) {
+		return true, "", nil
+	}
+
+	// Consumer groups work only for Kong Enterprise >=3.4.
+	infoSvc, ok := validator.AdminAPIServicesProvider.GetInfoService()
+	if !ok {
+		return true, "", nil
+	}
+	info, err := infoSvc.Get(ctx)
+	if err != nil {
+		validator.Logger.Debugf("failed to fetch Kong info: %v", err)
+		return false, ErrTextAdminAPIUnavailable, nil
+	}
+	version, err := kong.NewVersion(info.Version)
+	if err != nil {
+		validator.Logger.Debugf("failed to parse Kong version: %v", err)
+	} else {
+		kongVer := semver.Version{Major: version.Major(), Minor: version.Minor()}
+		if !version.IsKongGatewayEnterprise() || !kongVer.GTE(versions.ConsumerGroupsVersionCutoff) {
+			return false, ErrTextConsumerGroupUnsupported, nil
+		}
+	}
+
+	// This check forbids consumer group creation if the license is invalid or missing.
```

Comment:
🤔 To my eyes this only gets the service, doesn't it? There's no "check" in here that I can see. I believe this comment treats about the section below. Perhaps we could merge it with the comment in the line 200?